pkg/container/versioned/value_test.go

```diff
@@ -0,0 +1,388 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package versioned
+
+import (
+	"fmt"
+	"math/rand/v2"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/cilium/cilium/pkg/lock"
+)
+
+type testCleaner struct {
+	oldestVersion atomicVersion
+	cleanerMutex  lock.Mutex
+	cleanerCond   *sync.Cond
+}
+
+func newTestCleaner() *testCleaner {
+	c := &testCleaner{}
+	c.cleanerCond = sync.NewCond(&c.cleanerMutex)
+	return c
+}
+
+func (c *testCleaner) waitUntilCleaned(version version) {
+	c.cleanerMutex.Lock()
+	for oldest := c.oldestVersion.load(); version >= oldest; oldest = c.oldestVersion.load() {
+		fmt.Printf("Waiting due to oldest %d < %d version\n", oldest, version)
```

Comment:
nit: `<=`?

Reply:
This would be easier to read if the comparison order was flipped, i.e.:
```
  for ... ; oldest <= version; ... {
```
This would be read like "wait as long as the oldest (non-cleaned) version is less or equal than `version`", or "wait until oldest version is greater than `version`"; after this `version` is no longer available, it is "cleaned off".

Reply:
oh I just meant that the printed message is different from the actual check